src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -447,10 +455,14 @@ TokenizeSnippets::TokenizeSnippets() {
                 // Result op has a single input
                 internal_inputs.push_back(source_result->input_value(0));
             } else {
-                // We have to save explicitly FQ Constants to call ConstantFolding after Tokenization.
-                // After ConstantFolding we will move remaining non-scalar Constants from body using ConvertConstantsToParameters pass
+                // We need some non-scalar constants inside Subgraph in the following cases:
+                // [*] We have to save explicitly FQ Constants to call ConstantFolding after Tokenization.
+                //     After ConstantFolding we will move remaining non-scalar Constants from body using ConvertConstantsToParameters pass
+                // [*] We support Transpose with second Constant input (represents order). This Constant will not be scheduled
+                //     and will only be used to decompose Transpose into a proper Load, Store and Loop combination.
                 if ((utils::is_scalar_constant(input_node)) ||
```

Comment:
The same suggestion about `Constant` as common check

Reply:
Agreed, implemented